### PR TITLE
MongoDB Atlas: filters

### PR DESCRIPTION
**Pull request by anakin87**
part of #143

### CODE REVIEW

File: integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/errors.py [hunk 1]
```diff
@@ -1,4 +0,0 @@
-class MongoDBAtlasDocumentStoreError(Exception):
-    """Exception for issues that occur in a MongoDBAtlas document store"""
-
-    pass
```
Comment (anakin87) [inline integrations/mongodb_atlas/src/haystack_integrations/document_stores/mongodb_atlas/errors.py:4]: it was unused

File: integrations/mongodb_atlas/tests/test_document_store.py [hunk 2]
```diff
@@ -8,42 +8,43 @@
...
+from pymongo.driver_info import DriverInfo
 
 
 @pytest.mark.skipif(
     "MONGO_CONNECTION_STRING" not in os.environ,
     reason="No MongoDB Atlas connection string provided",
 )
-class TestDocumentStore(CountDocumentsTest, WriteDocumentsTest, DeleteDocumentsTest):
+@pytest.mark.integration
+class TestDocumentStore(DocumentStoreBaseTests):
```
Comment (anakin87) [inline integrations/mongodb_atlas/tests/test_document_store.py:24]: I included all the base tests about filters.

**Review by julian-risch — CHANGES_REQUESTED** [review]
Looks very good to me already. I have one comment on a code block that we re-use multiple times. Should become a separate method for input parameter validation. The PR can be merged right after that is fixed. 👍

**Review by julian-risch — APPROVED** [review]
LGTM! 👍

**Pull request merged**